lightning/src/util/test_utils.rs

```diff
@@ -856,15 +857,80 @@ impl<Signer: sign::ecdsa::EcdsaChannelSigner> Persist<Signer> for TestPersister
 	}
 }
 
+type SPSCKVChannelState = Arc<Mutex<(Option<Result<(), io::Error>>, Option<Waker>)>>;
+struct SPSCKVChannel(SPSCKVChannelState);
+impl Future for SPSCKVChannel {
+	type Output = Result<(), io::Error>;
+	fn poll(self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), io::Error>> {
+		let mut state = self.0.lock().unwrap();
+		state.0.take().map(|res| Poll::Ready(res)).unwrap_or_else(|| {
+			state.1 = Some(cx.waker().clone());
+			Poll::Pending
+		})
+	}
+}
+
 pub struct TestStore {
+	pending_async_writes: Mutex<HashMap<String, Vec<(usize, SPSCKVChannelState, Vec<u8>)>>>,
 	persisted_bytes: Mutex<HashMap<String, HashMap<String, Vec<u8>>>>,
 	read_only: bool,
 }
 
 impl TestStore {
 	pub fn new(read_only: bool) -> Self {
+		let pending_async_writes = Mutex::new(new_hash_map());
 		let persisted_bytes = Mutex::new(new_hash_map());
-		Self { persisted_bytes, read_only }
+		Self { pending_async_writes, persisted_bytes, read_only }
+	}
+
+	pub fn list_pending_async_writes(
+		&self, primary_namespace: &str, secondary_namespace: &str, key: &str,
+	) -> Vec<usize> {
+		let key = format!("{primary_namespace}/{secondary_namespace}/{key}");
+		let writes_lock = self.pending_async_writes.lock().unwrap();
+		writes_lock
+			.get(&key)
+			.map(|v| v.iter().map(|(id, _, _)| *id).collect())
+			.unwrap_or(Vec::new())
+	}
+
+	pub fn complete_async_writes_through(
+		&self, primary_namespace: &str, secondary_namespace: &str, key: &str, write_id: usize,
+	) {
+		let prefix = format!("{primary_namespace}/{secondary_namespace}");
+		let key = format!("{primary_namespace}/{secondary_namespace}/{key}");
+
+		let mut persisted_lock = self.persisted_bytes.lock().unwrap();
+		let mut writes_lock = self.pending_async_writes.lock().unwrap();
+
+		let pending_writes = writes_lock.get_mut(&key).unwrap();
+		pending_writes.retain(|(id, res, data)| {
+			if *id <= write_id {
+				let namespace = persisted_lock.entry(prefix.clone()).or_insert(new_hash_map());
+				*namespace.entry(key.to_string()).or_default() = data.clone();
+				let mut future_state = res.lock().unwrap();
+				future_state.0 = Some(Ok(()));
+				if let Some(waker) = future_state.1.take() {
+					waker.wake();
+				}
+				false
+			} else {
+				true
+			}
+		});
+	}
+
+	pub fn complete_all_async_writes(&self) {
```

Comment:
Comments explaining this function and the one above would be nice. And some general directions on how to use the test store.